packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/App.kt

```diff
@@ -84,6 +84,19 @@ public interface App {
      */
     public val sync: Sync
 
+    /**
+     * Current base URL to communicate with App Services.
+     */
+    public val baseUrl: String
+
+    /**
+     * Sets the App Services base url. Changing the URL would trigger a client reset.
+     *
+     * @param baseUrl The new App Services base url. Use [AppConfiguration.DEFAULT_BASE_URL] to set it
+     * to the default value.
+     */
+    public suspend fun updateBaseUrl(baseUrl: String)
```

Comment:
Passing an empty string to `updateBaseUrl` would result in the base URL being set to a default value set in core. 

Because the default URL is already set in core, we could replace the actual URL value in the SDK with an empty string. This way we could avoid maintaining the URL on the SDK. 

```
AppConfiguration.DEFAULT_BASE_URL = ""
```

We would address this on a different PR.

Reply:
We could also make `baseUrl` optional and allow users to supply `null` here to the same effect of supplying empty string. I feel this is a bit more obvious and the C API already works that way.